pkg/authorization/api/v1beta1/types.go

```diff
@@ -16,15 +16,15 @@ import (
 // PolicyRule holds information that describes a policy rule, but does not contain information
 // about who the rule applies to or which namespace the rule applies to.
 type PolicyRule struct {
-	// Deny is true if any request matching this rule should be denied.  If false, any request matching this rule is allowed.
-	Deny bool `json:"deny"`
 	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  VerbAll represents all kinds.
 	Verbs []string `json:"verbs"`
 	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.
 	// If the Authorizer does not recognize how to handle the AttributeRestrictions, the Authorizer should report an error.
 	AttributeRestrictions kruntime.RawExtension `json:"attributeRestrictions"`
-	// ResourceKinds is a list of kinds this rule applies to.  ResourceAll represents all kinds.
-	ResourceKinds []string `json:"resourceKinds""`
+	// ResourceKindsDeprecated is a list of resources this rule applies to.  ResourceAll represents all resources.
+	ResourceKindsDeprecated []string `json:"resourceKinds""`
```

Comment:
Without having  `ResourceKindsDeprecated` mapping to the old `resourceKinds` field in the json, how do I get access to the `resourceKinds` fields from a previously serialized value in order to have my conversion functions propagate the value to the in-memory representation?